docs/example/saxml/leader-worker-set.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: leaderworkerset.x-k8s.io/v1
+kind: LeaderWorkerSet
+metadata:
+  name: saxml-multi-host
+  annotations:
+    leaderworkerset.sigs.k8s.io/exclusive-topology: cloud.google.com/gke-nodepool
+spec:
+  replicas: 2
+  leaderWorkerTemplate:
+    size: 8
+    restartPolicy: RecreateGroupOnPodRestart
+    leaderTemplate:
+      metadata:
+        labels:
+            app: leader-pod
+      spec:
+        serviceAccountName: sax-sa
+        nodeSelector:
+          cloud.google.com/gke-tpu-topology: 4x8
+          cloud.google.com/gke-tpu-accelerator: tpu-v5-lite-podslice
+        containers:
+        - name: sax-admin-server
+          image: us-docker.pkg.dev/cloud-tpu-images/inference/sax-admin-server:nightly
+          securityContext:
+            privileged: true
+          ports:
+          - containerPort: 10000
+          env:
+          - name: SAX_REPLICA_NAME
+            valueFrom:
+              fieldRef: 
+                fieldPath: metadata.name 
+          - name: SAX_GCS_BUCKET
+            value: BUCKET_NAME
+        - name: sax-model-server
+          image: us-docker.pkg.dev/cloud-tpu-images/inference/sax-model-server:nightly
+          args: ["--port=10001", "--platform_chip=tpuv5e"]
+          ports:
+          - containerPort: 10001
+          - containerPort: 8471
+          securityContext:
+            privileged: true
+          env:
+          - name: SAX_REPLICA_NAME
+            valueFrom:
+              fieldRef: 
+                fieldPath: metadata.name 
+          - name: SAX_GCS_BUCKET
+            value: BUCKET_NAME
+          resources:
+            requests:
+              google.com/tpu: 4
+            limits:
+              google.com/tpu: 4
+        - name: sax-http
+          image: gcr.io/tpu-vm-gke-testing/saxml-lws-testing-modified/httpserver-merged:latest
+          volumeMounts:
+            - name: publish-config
+              mountPath: /publish/config 
+          ports:
+          - containerPort: 8888
+          env:
+          - name: SAX_REPLICA_NAME
+            valueFrom:
+              fieldRef: 
+                fieldPath: metadata.name 
+          - name: SAX_GCS_BUCKET
+            value: BUCKET_NAME
+        volumes:
+        - name: publish-config
+          configMap:
+            name: configmaplws
+    workerTemplate:
+      spec:
+        serviceAccountName: sax-sa
+        nodeSelector:
+          cloud.google.com/gke-tpu-accelerator: tpu-v5-lite-podslice
+          cloud.google.com/gke-tpu-topology: 4x8
```

Comment:
Removed

Reply:
Just tested it out, if I remove it the worker pods do not get assigned.

Reply:
No errors show up. The worker pods do not show up as a workload at all. I'm guessing it doesn't know where to assign it because of the lack of node selector.

Reply:
So the worker pods don't get created? Does the workers StatefulSet get created? This is very strange and we need to debug it because it may indicate a bug in LWS controller.

Reply:
The workers StatefulSet does not get created. The LWS controller gets this error
``` Reconciler error	{"controller": "pod", "controllerGroup": "", "controllerKind": "Pod", "Pod": {"name":"saxml-multi-host-0","namespace":"default"}, "namespace": "default", "name": "saxml-multi-host-0", "reconcileID": "b1311d1a-bd39-460c-a99f-7fa2ac19bd06", "error": "admission webhook \"warden-validating.common-webhooks.networking.gke.io\" denied the request: GKE Warden rejected the request because it violates one or more constraints.\nViolations details: {\"[denied by tpu-accelerator-topology-constraints]\":[\"Missing nodeSelector/nodeAffinity label cloud.google.com/gke-tpu-accelerator. Please specify cloud.google.com/gke-tpu-accelerator when requesting tpu resources.\",\"Missing nodeSelector/nodeAffinity label cloud.google.com/gke-tpu-topology. Please specify cloud.google.com/gke-tpu-accelerator and cloud.google.com/gke-tpu-topology requesting tpu resources.\" ```

Reply:
ah, ok, so on GKE it must be set it, GKE warden enforces it.